### PR TITLE
feat(index): support building inverted index for the field column on Mito

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Respect field columns when building the index. See the `do_update` function in `src/mito2/src/sst/index/inverted_index/creator.rs`

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/mito2/src/read/scan_region.rs [hunk 1]
```diff
@@ -393,20 +393,29 @@ impl ScanRegion {
             .and_then(|c| c.index_cache())
             .cloned();
 
+        // TODO(zhongzc): currently we only index tag columns, need to support field columns.
```
Comment (waynexia) [inline src/mito2/src/read/scan_region.rs:396]: This comment doesn't seem to match the PR title 🤔

File: src/mito2/src/sst/index/inverted_index/creator.rs [hunk 2]
```diff
@@ -210,6 +203,32 @@ impl InvertedIndexer {
...
         }
 
+        for field in batch.fields() {
+            if !self.indexed_column_ids.contains(&field.column_id) {
+                continue;
+            }
+
+            let sort_field = SortField::new(field.data.data_type());
+            let col_id_str = field.column_id.to_string();
+            for i in 0..n {
```
Comment (waynexia) [inline src/mito2/src/sst/index/inverted_index/creator.rs:213]: I'm a little worried about the overhead for this loop 😢

**Review by waynexia — APPROVED** [review]
LGTM in general

File: src/mito2/src/read/scan_region.rs (hunk 1 above)
Comment (zhongzc) [reply src/mito2/src/read/scan_region.rs:396]: It's not completely assembled, but this is the key entrance and everything inside is ready.

File: src/mito2/src/sst/index/inverted_index/creator.rs (hunk 2 above)
Comment (zhongzc) [reply src/mito2/src/sst/index/inverted_index/creator.rs:213]: which part? `to_string` or encoding?

Comment (v0y4g3r) [reply src/mito2/src/sst/index/inverted_index/creator.rs:213]: > I'm a little worried about the overhead for this loop 😢

Looks like this loop lieterally iterates and builds index for each row?

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**